commons-core/src/main/scala/com/avsystem/commons/serialization/GenCodec.scala

```diff
@@ -370,7 +416,15 @@ object GenCodec extends RecursiveAutoCodecs with TupleGenCodecs {
   }
 
   implicit def arrayCodec[T: ClassTag : GenCodec]: GenCodec[Array[T]] =
-    nullableList[Array[T]](_.iterator(read[T]).toArray[T], (lo, arr) => arr.iterator.writeToList(lo))
+    nullableList[Array[T]](_.iterator(read[T]).toArray[T], (lo, arr) => {
+      lo.declareSize(arr.length)
+      @tailrec def loop(idx: Int): Unit =
```

Comment:
Isn't this premature optimization vs `arr.foreach`? I remember the differences in performance were marginal when I last measured it.

Reply:
Well, it's just a loop. It wasn't specifically intended as optimization.